lib/wallet/api/http/Cardano/Wallet/Api/Http/Server/Error.hs

```diff
@@ -526,11 +528,17 @@ instance IsServerError ErrBalanceTx where
 
 instance IsServerError ErrBalanceTxInternalError where
     toServerError = \case
-        ErrUnderestimatedFee coin _st ->
+        ErrUnderestimatedFee coin candidateTx (KeyWitnessCount nWits nBootWits) ->
             apiError err500 BalanceTxUnderestimatedFee $ T.unwords
                 [ "I have somehow underestimated the fee of the transaction by"
                 , pretty coin
                 , "and cannot finish balancing."
+                , "I have assumed the transaction needs" <> T.pack (show nWits)
+                , "verification key witnesses and" <> T.pack (show nBootWits)
+                , "bootstrap witnesses."
+                , "\n\n"
+                , "The candidate tx is:"
+                , T.pack (show candidateTx)
```

Comment:
Perhaps here we could also create a rich JSON error object instead of trying to embed all the information in a single string!

Reply:
Done as part of #3766